upb/json/parser.rl

```diff
@@ -1828,54 +1818,18 @@ static void end_object(upb_json_parser *p) {
   }
 }
 
-static bool is_double_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kDoubleValueFullMessageName) == 0;
-}
-
-static bool is_float_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kFloatValueFullMessageName) == 0;
-}
-
-static bool is_int64_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kInt64ValueFullMessageName) == 0;
-}
-
-static bool is_uint64_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kUInt64ValueFullMessageName) == 0;
-}
-
-static bool is_int32_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kInt32ValueFullMessageName) == 0;
-}
-
-static bool is_uint32_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kUInt32ValueFullMessageName) == 0;
-}
-
-static bool is_bool_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kBoolValueFullMessageName) == 0;
-}
-
-static bool is_string_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kStringValueFullMessageName) == 0;
-}
-
-static bool is_bytes_value(const upb_msgdef *m) {
-  return strcmp(upb_msgdef_fullname(m), kBytesValueFullMessageName) == 0;
-}
-
 static bool is_number_wrapper(const upb_msgdef *m) {
-  return is_double_value(m) ||
-         is_float_value(m) ||
-         is_int64_value(m) ||
-         is_uint64_value(m) ||
-         is_int32_value(m) ||
-         is_uint32_value(m);
+  return (upb_msgdef_wellknowntype(m) == UPB_WELLKNOWN_DOUBLEVALUE) ||
```

Comment:
Could we put all of the wrapper types in a contiguous range so that this can just be something like (type >= UPB_WELLKNOWN_FIRSTWRAPPER && type <= UPB_WELLKNOWN_LASTWRAPPER)?

Also we should only call upb_msg_wellknowntype(m) once at a time, and store the value.